docs/source/developers/index.md

```diff
@@ -121,6 +121,85 @@ your new provider's `id`:
 [LLM]: https://api.python.langchain.com/en/v0.0.339/llms/langchain.llms.base.LLM.html#langchain.llms.base.LLM
 [BaseChatModel]: https://api.python.langchain.com/en/v0.0.339/chat_models/langchain.chat_models.base.BaseChatModel.html
 
+#### Configuration for custom providers
+
+You can add custom fields into the settings dialogue for your custom model by specifying a list of fields as shown
+below.
+
+These will be passed into the `__init__` as kwargs, with the key specified by the key in the field object.
+
+The label specified in the field object determines the text shown in the configuration section of the user interface.
+
+```python
+from jupyter_ai_magics import BaseProvider
+from jupyter_ai_magics.providers import TextField, MultilineTextField
+from langchain_community.llms import FakeListLLM
+
+
+class MyProvider(BaseProvider, FakeListLLM):
+    id = "my_provider"
+    name = "My Provider"
+    model_id_key = "model"
+    models = [
+        "model_a",
+        "model_b"
+    ]
+
+    fields: ClassVar[List[Field]] = [
+        TextField(key="my_llm_parameter", label="The name for my_llm_parameter to show in the UI"),
+        MultilineTextField(key="custom_config", label="Custom Json Config", format="json"),
+    ]
+
+    def __init__(self, **kwargs):
+        model = kwargs.get("model_id")
+        kwargs["responses"] = (
+            ["This is a response from model 'a'"]
+            if model == "model_a" else
+            ["This is a response from model 'b'"]
+        )
+        super().__init__(**kwargs)
+
+```
+
+#### Api key  for custom providers
```

Comment:
Since both of these sections use the same example, it seems like these can be merged. Can you merge these 2 sections into a single section titled "API keys and fields for custom providers"?

Reply:
Will do as soon as I get chance! :+1: 
First however, there are some cakes which must be baked.

Reply:
Updated as requested, merging the two sections & examples into one.  :slightly_smiling_face: